src/System.ComponentModel.TypeConverter/tests/ContextStackTests.cs

```diff
@@ -0,0 +1,204 @@
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+
+using System;
+using System.ComponentModel.Design.Serialization;
+using Xunit;
+
+namespace System.ComponentModel.Tests
+{
+	
+	public class ContextStackTest
+	{
+		[Fact]
+		public void IntegrityTest ()
+		{
+			ContextStack stack = new ContextStack ();
+
+			string one = "one";
+			string two = "two";
+			stack.Push (two);
+			stack.Push (one);
+			Assert.Same (one, stack [typeof (string)]);
+			Assert.Same (one, stack [0]);
+			Assert.Same (one, stack.Current);
+
+			Assert.Same (one, stack.Pop ());
+
+			Assert.Same (two, stack [typeof (string)]);
+			Assert.Same (two, stack [0]);
+			Assert.Same (two, stack.Current);
+
+			string three = "three";
+			stack.Append (three);
+
+			Assert.Same (two, stack[typeof (string)]);
+			Assert.Same (two, stack[0]);
+			Assert.Same (two, stack.Current);
+
+			Assert.Same (two, stack.Pop ());
+
+			Assert.Same (three, stack[typeof (string)]);
+			Assert.Same (three, stack[0]);
+			Assert.Same (three, stack.Current);
+			Assert.Same (three, stack.Pop ());
+
+			Assert.Null (stack.Pop ());
+			Assert.Null (stack.Current);
+		}
+
+		[Fact]
+		public void Append_Context_Null ()
+		{
+			ContextStack stack = new ContextStack ();
+			try {
+				stack.Append (null);
+				Assert.False(true);
+			} catch (ArgumentNullException ex) {
+				Assert.Equal (typeof (ArgumentNullException), ex.GetType ());
+				Assert.Null (ex.InnerException);
+				Assert.NotNull (ex.Message);
+				Assert.Equal ("context", ex.ParamName);
+			}
+		}
+
+		[Fact] // Item (Int32)
+		public void Indexer1 ()
+		{
+			ContextStack stack = new ContextStack ();
+			string one = "one";
+			string two = "two";
+
+			stack.Push (one);
+			stack.Push (two);
+
+			Assert.Same (two, stack [0]);
+			Assert.Same (one, stack [1]);
+			Assert.Null (stack [2]);
+			Assert.Same (two, stack.Pop ());
+			Assert.Same (one, stack [0]);
+			Assert.Null (stack [1]);
+			Assert.Same (one, stack.Pop ());
+			Assert.Null (stack [0]);
+			Assert.Null (stack [1]);
+		}
+
+		[Fact] // Item (Int32)
+		public void Indexer1_Level_Negative ()
+		{
+			ContextStack stack = new ContextStack ();
+			stack.Push (new Foo ());
+
+			try {
+				object context = stack [-1];
+				Assert.False(true);
+			} catch (ArgumentOutOfRangeException ex) {
+				Assert.Equal (typeof (ArgumentOutOfRangeException), ex.GetType ());
+				Assert.Null (ex.InnerException);
+				Assert.Equal (new ArgumentOutOfRangeException ("level").Message, ex.Message);
+				Assert.Equal ("level", ex.ParamName);
+			}
+
+			try {
+				object context = stack [-5];
+				Assert.False(true);
+			} catch (ArgumentOutOfRangeException ex) {
```

Comment:
All of the tests that follow this pattern should be switched to use Assert.Throws